### DISCUSSION TITLE
Taxonomy and layout discussion

### DISCUSSION THREAD

**Discussion by mattstratton** [devopsdays/raging-yak#5] [General]
Let's use this to think about overall layout etc

First topic! 

Maybe we should rethink the taxonomy. We tend to think about the grouping a yearly basis, but what if...the taxonomy was city oriented?

So you end up with a structure like this:
```
content/
└── events
    └── ponyville
        └── 2023
            ├── index.md
            ├── logo.png
            ├── speakers.md
            └── speaker-images
                └── twilight-sparkle.jpg
        └── 2024
            ├── index.md
            └── speakers.md
    └── hoofington
        └── 2024
            ├── index.md
            ├── logo.png
            ├── speakers.md
            └── speaker-images
                └── twilight-sparkle.jpg
        └── 2025
            ├── index.md
            └── speakers.md
```

**Comment by nimbinatus** [comment]
Hm! That might be useful, especially if then we kept the "archived" pages under that. I was thinking that would be a very easy way to then tell programmatically that a page is "old" and gets a banner (maybe the creation script adds a tag to the metadata in the data.yml file that reveals the banner?). And if we did it right, we could even have the sponsor logos be kept static for each event.

Maybe
```
assets/
└── events
    └── ponyville
        └── 2023
            ├── logo.png
            └── speakers
                └── twilight-sparkle.jpg
            └── sponsors
                └── company.jpg
    └── hoofington
        └── 2024
            ├── logo.png
            └── speakers
                └── twilight-sparkle.jpg
            └── sponsors
                └── company.jpg
content/
└── events
    └── ponyville
        └── 2023
            ├── index.md
            ├── speakers.md
        └── 2024
            ├── index.md
            └── speakers.md
    └── hoofington
        └── 2024
            ├── index.md
            ├── speakers.md
        └── 2025
            ├── index.md
            └── speakers.md
```

**Reply by mattstratton** [reply]
>we could even have the sponsor logos be kept static for each event

This will end up with a large amount of duplicate files which will make the repo even WORSE

(the old webby implementation worked this way, and it was pretty bad, and that was only for four years of devopsdays!)

What i'm wondering about (for the "generated static versions") if there's a way to sorta eliminate them from the standard build, but you could do them on demand (this is a different topic but it just occurred to me, and maybe I should start a new thread on it, but gonna get my ideas out briefly):

Like, once an event is "archived", it gets marked "draft" in hugo (or some other type of thing). So a normal hugo build will NOT build those pages. Which is fine because the static HTML exists in the `static` directory.

BUT...if you wanted (some type of maint job or something) you COULD run a version of the build (with drafts enabled, if that's the way it works) which would then created UPDATED versions of those files, which would be copied to static, etc.

Basically, this allows you to have only current stuff be "built" every time, but keeping the ability to refresh the archived stuff. Hmm.

**Reply by nimbinatus** [reply]
:thinking:  I was kinda wondering if there was a way to snapshot like that. What if the build looks for a tag in the frontmatter like "current" before building? Something we could change with whatever system comes in for creating a new event?

**Reply by nimbinatus** [reply]
Once there's no more "current" tag, it adds a banner and preserves it. And if we did a basic city landing page, we wouldn't need to have each year on each event's page, which removes the need to build it each year. Just always link back to the city page...

**Reply by mattstratton** [reply]
That’s why I said use “draft” because it’s built into Hugo so we don’t have to build our own logic and it just is a flag to the command to CHOOSE to include them when building!

**Reply by mattstratton** [reply]
But that’s backwards logic from your proposal (mine is flag everything BUT current, etc) but it’s the most error proof for folks running the build to not accidentally include old stuff etc. 

Also much less logic built into our code that now has to think about “do I build or not?” on every page in our execution vs the far more optimized bc in the Hugo binary version.

**Reply by mattstratton** [reply]
One thing you will discover with this site, is that we hit death by 1000 cuts a lot. Because it’s SO MANY PAGES. So adding even small logic increases the build exponentially :/

**Reply by mattstratton** [reply]
> Once there's no more "current" tag, it adds a banner and preserves it. And if we did a basic city landing page, we wouldn't need to have each year on each event's page, which removes the need to build it each year. Just always link back to the city page...

Yeah the nice thing about a “container” page for the city overall is that meta nav gets solved. 

Now, let me tell you the wrinkle in all these …. `event_group` - there are cities who are different geo cities but get grouped together for nav because their the same “event” just in different city locales each year (Sydney/Melboune does this, for example)
